### PR TITLE
Add configurable break templates

### LINKED ISSUE #192: Break templates
## Summary
Introduce configurable break templates for faster break selection.

## Scope
- Implement the primary workflow and data-path changes for this feature.
- Update CLI/TUI/docs/tests where applicable.

## Acceptance criteria
- Behavior is available and documented as described.
- Existing workflows remain backward compatible unless explicitly changed.

### PR DESCRIPTION
**Pull request by utilForever**
## What

Add global named break templates and a fast selection flow in the profile manager so users can switch break settings quickly. Persist template data in config, expose selected and available template context in CLI profile/status outputs, and document the workflow.

## Why

Issue #192 requests configurable break templates to reduce repetitive manual break-duration edits. This change adds the template workflow while keeping existing profile and automation behavior backward compatible.

### Required

- [x] `cargo check --all` passes
- [x] `cargo fmt --all -- --check` passes
- [x] `cargo clippy --all-targets -- -D warnings` passes
- [x] `cargo test --all` passes
- [x] I linked the related issue (for example: `Closes #25`) - Closes #192

### Functional Validation

- [x] Pomodoro timer behavior was verified for this change (if applicable)
- [ ] Site blocking behavior was verified for this change (if applicable) - N/A
- [ ] WakaTime integration behavior was verified for this change (if applicable) - N/A
- [x] I added or updated tests for changed behavior (if applicable)

### Configuration & Docs

- [x] User-facing docs were updated (`README.md` or relevant docs, if applicable)
- [x] New dependencies/configuration are documented (if applicable)
- [x] No sensitive values or credentials were introduced

### If Applicable

- [ ] Security impact considered (run `cargo audit` locally if needed) - N/A
- [ ] Breaking behavior changes are clearly described in this PR - N/A

## Summary by CodeRabbit

* **New Features**
  * Break templates: cycle between predefined break configurations using `[` / `]` from timer view
  * Included templates: "Classic" and "Deep Work" with configurable short/long breaks and long-break intervals
  * Selected template persists; Profile Manager shows active template name and summary

* **CLI**
  * Status and profile outputs show the selected template and list available templates

* **Documentation**
  * README documents break-template usage and example config entries

* **Tests**
  * Status JSON tests updated to require template fields

### CODE REVIEW

File: src/app.rs [hunk 1]
```diff
@@ -531,6 +541,9 @@ impl App {
         let blocklist_profiles = config.blocklist_profiles.clone();
         let active_blocklist_profile =
             blocklist_profile_index(&blocklist_profiles, &config.selected_blocklist_profile);
+        let break_templates = config.break_templates.clone();
+        let active_break_template =
+            break_template_index(&break_templates, &config.selected_break_template);
```
[the remark this answers is not in the document]
Comment (utilForever) [reply src/app.rs]: Good catch. I updated break-template restoration so selection stays aligned with effective custom break values. On load and after profile edit commits, the app now recomputes the active template from the custom short/long break and cadence values, and clears persisted selection when no template matches.

File: src/cli.rs [hunk 2]
```diff
@@ -3113,6 +3131,38 @@ fn profile_id(profile: ProfileId) -> &'static str {
...
+        .break_templates
+        .iter()
+        .find(|template| {
+            template
+                .name
+                .eq_ignore_ascii_case(&config.selected_break_template)
+        })
+        .map(break_template_view)
+        .or_else(|| config.break_templates.first().map(break_template_view))
+        .unwrap_or_else(|| break_template_view(&BreakTemplateConfig::default()))
```
Comment (utilForever) [inline src/cli.rs]: Fixed. selected_break_template_view now trims selected_break_template before the case-insensitive lookup, so surrounding whitespace no longer prevents matching the intended template.

File: src/ui.rs [hunk 3]
```diff
@@ -738,12 +738,18 @@ fn render_profile_manager(frame: &mut Frame, app: &App) {
...
+        app.active_break_template_name(),
+        app.active_break_template_summary()
+    ))
+    .style(
+        Style::default()
+            .fg(Color::White)
+            .add_modifier(Modifier::BOLD),
+    )
+    .wrap(Wrap { trim: true });
     frame.render_widget(current, inner[0]);
```
[the remark this answers is not in the document]
Comment (utilForever) [reply src/ui.rs:753]: Fixed. I increased the profile-manager header region from 1 to 2 rows so the longer profile plus break-template line can wrap instead of clipping on narrow terminals.

File: src/app.rs [hunk 4]
```diff
@@ -3720,6 +3834,86 @@ impl App {
...
+        self.clamp_break_template_selection();
+        if self.break_templates.is_empty() {
+            return;
+        }
+        let current = self
+            .active_break_template
+            .unwrap_or(0)
+            .min(self.break_templates.len().saturating_sub(1));
+        let next = (current + 1) % self.break_templates.len();
+        self.switch_break_template(next);
```
[the remark this answers is not in the document]
Comment (utilForever) [reply src/app.rs:3871]: Fixed. I updated template cycling so an unlinked state is treated as a real sentinel: ] now selects the first template, and [ selects the last template. I also added a regression test covering both directions from ctive_break_template = None.

File: src/cli.rs [hunk 5]
```diff
@@ -3113,6 +3131,35 @@ fn profile_id(profile: ProfileId) -> &'static str {
...
+
+fn selected_break_template_view(config: &AppConfig) -> BreakTemplateView {
+    let selected_name = config.selected_break_template.trim();
+    config
+        .break_templates
+        .iter()
+        .find(|template| template.name.eq_ignore_ascii_case(selected_name))
+        .map(break_template_view)
+        .or_else(|| config.break_templates.first().map(break_template_view))
+        .unwrap_or_else(|| break_template_view(&BreakTemplateConfig::default()))
```
[the remark this answers is not in the document]
Comment (utilForever) [reply src/cli.rs]: Fixed. CLI selected template resolution now follows the same name-plus-timings rules as the app runtime. If no template matches effective custom break timings, it no longer falls back to the first template; instead it reports a Custom sentinel with the effective custom short/long/cadence values.

File: src/config.rs [hunk 6]
```diff
@@ -47,6 +47,12 @@ pub struct AppConfig {
     /// When this is absent, the app derives it from the legacy duration fields.
     #[serde(default)]
     pub custom_profile: Option<CustomProfileConfig>,
+    /// Globally reusable break templates for quick selection.
+    #[serde(default = "default_break_templates")]
+    pub break_templates: Vec<BreakTemplateConfig>,
+    /// Name of the active break template.
+    #[serde(default = "default_break_template_name")]
+    pub selected_break_template: String,
```
[the remark this answers is not in the document]
Comment (utilForever) [reply src/config.rs:55]: Fixed. selected_break_template now uses an empty serde default (#[serde(default)]), so legacy configs without this field deserialize as empty instead of Classic. AppConfig::default() still seeds Classic for brand-new configs only.

File: src/config.rs [hunk 7]
```diff
@@ -721,6 +793,9 @@ impl AppConfig {
             default_break_glass_duration_secs(),
         );
         self.custom_profile = self.custom_profile.map(|profile| profile.normalized());
+        self.break_templates = normalize_break_templates(&self.break_templates);
+        self.selected_break_template =
+            normalize_selected_break_template(&self.selected_break_template, &self.break_templates);
```
[the remark this answers is not in the document]
Comment (utilForever) [reply src/config.rs]: Fixed. I updated config normalization to resolve selected break templates against effective custom break timings, not by first-template fallback. It now keeps a selected name only when name+timings match, falls back to a timing match when available, and otherwise clears the selection so unrelated saves cannot persist a bogus template link.
